osu.Game/Screens/Play/HUD/KeyCounterDisplay.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input.Events;
+using osu.Game.Configuration;
+using osuTK;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    /// <summary>
+    /// A flowing display of all gameplay keys. Individual keys can be added using <see cref="InputTrigger"/> implementations.
+    /// </summary>
+    public abstract partial class KeyCounterDisplay : CompositeDrawable
+    {
+        /// <summary>
+        /// Whether the key counter should be visible regardless of the configuration value.
+        /// This is true by default, but can be changed.
+        /// </summary>
+        public Bindable<bool> AlwaysVisible { get; } = new Bindable<bool>(true);
+
+        /// <summary>
+        /// The <see cref="KeyCounter"/>s contained in this <see cref="KeyCounterDisplay"/>.
+        /// </summary>
+        public abstract Container<KeyCounter> Counters { get; }
```

Comment:
This being a `Container<KeyCounter>` again means that none of the last 3 commits actually changed anything in the state of affairs, because if it's exposed as a container, then you can add anything you like again through it.

Can we revert 5d15426c275084144d68e8790f0275d985c52269? I was honestly quite fine with the `IEnumerable<KeyCounter>`. I don't find it at all messy. The only consumers of this property are: test scenes, and `RulesetInputManager`. It's very contained.

Reply:
The purpose of that commit was to fix this
https://github.com/ppy/osu/actions/runs/4360048022
Does that mean that I need to remove the checks for the counter flows visibility in `TestSceneHUDOverlay`? This is the only way I see to solve the failures (plus I think that this check will most likely be irrelevant as new variants get implemented)

Reply:
How does changing that from `IEnumerable` to `Container` fix test failures? That's an alarm bell if you ask me. Just changing the *returned type* fix it? Can you explain why?

Reply:
Because of this??
https://github.com/ppy/osu/blob/8f6df5ea0f7f721c630fc8cad93bb3eef869d1d9/osu.Game.Tests/Visual/Gameplay/TestSceneHUDOverlay.cs#L48

Reply:
Acutally the more I look into it, the more it feels weird to me

Reply:
Well this is where I would just fix the test to not use `.ChildrenOfType<>()`. But maybe it's just me.

Reply:
I'll get into this later, I'm really not focused right now.